mlir/utils/jenkins/MIOpenDriver.py

```diff
@@ -595,10 +603,35 @@ def main(args=None):
         nargs='*',
         help="The specific config to test, if you want to test one"
     )
+
+    parser.add_argument(
+        "--rocmlir_gen_flags",
+        type=str,
+        default=argparse.SUPPRESS,
+        help="rocmlir-gen flags to toggle each feature"
+    )
+
     parsed_args = parser.parse_args(args)
 
-    # Impose default behavior when no args have been passed
-    if len(args) == 0:
+    args_len = len(args)
+    # If rocmlir_gen_flags not provided, infer it from the chip
+    rocmlir_gen_flags = ''
```

Comment:
Let me clarify:
1. rocmlir_gen_flags is not related to this args_len business: the point of args_len is to default to compaliring to MIOpen and MLIR if you call the script with no options
2. You must not infer default values for rocmlir_gen_flags. Don't do this switch-on-arch business. What you should do with rocmlir_gen_flags is split it on whitespace and append those split values to the end of the arguments to rocmlir_gen. rocmlir_gen should be inferring from `arch` by default, but I want rocmlir_gen_flags specifically to allow me to do run, on a gfx908, `./bin/MIOpenDriver.py -b --rocmlir_gen_flags="-mfma=off"` to get non-xdlops perf numbers

Reply:
1. So `MIOpenDriver.py --rocmlir_gen_flags="-mfma=off"` will never happen? which means when comparing MIOpen and MLIR you always want to infer the features from the arch?

Reply:
No, it could happen, and it has the semantics "compare MIOpen and MLIR, when invoking `rocmlir-gen` add the flags "-mfma=off" to the command line"

It's a somewhat silly thing to ask for, and if someone actually wants to do that exact comparison, they'll probably need to patch in not setting the solver to the Xdlops version on the MIOpen side.

Reply:
`--rocmlir_gen_flags` is the very simple option "Trust me, I want rocmlir-gen to get some extra arguments, just do it." This is by analogy to, say, `CFLAGS`

Reply:
If that is the case, `MIOpenDriver.py --rocmlir_gen_flags='-mfma=off'` will not set `batch_both` unless we rule out `--rocmlir_gen_flags` from `args_len`.

Reply:
... good point

Reply:
`./bin/MIOpenDriver.py --rocmlir_gen_flags="-mfma=off"` is an error because you didn't specify a mode

Reply:
... but if you go and add `--batch_both` ... I'll assume you know what you want